### PR TITLE
Allow user to specify which fuzzy finder to use

### PR DESCRIPTION
**Pull request by khuei**
[Good to merge] This patch is completed.

I just want to see if you want to include this feature in your project before I keep working on this.

### CODE REVIEW

**Comment by filiparag** [comment]
I like the idea and your code changes follow the style.

Just one question: are other fuzzy finders compatible with parameters for `fzf`?

**Comment by khuei** [comment]
> Just one question: are other fuzzy finders compatible with parameters for `fzf`?

I'm not sure about all the other fuzzy finders, so I will implement a check for it just in case more fuzzy finders popping up in the future.

The only other fuzzy finder I use is [`skim`](https://github.com/lotabout/skim), which is also compatible with `fzf`'s parameters.

**Comment by khuei** [comment]
@filiparag You can review this patch now.

Thank you in advance.

File: wikiman.sh [hunk 1]
```diff
@@ -285,6 +296,23 @@ while getopts l:s:H:pqhRSk o; do
...
+		exit 1
+	}
+done
+
+# Check if fuzzy finder compatible with used fzf's parameters
+
+[ "$conf_fuzzy_finder" = "fzf" ] || [ "$conf_fuzzy_finder" = "sk" ] || {
+	echo "Error: $conf_fuzzy_finder is not compatible with the paramters used in this script"
+	exit 1
+}
```
Comment (filiparag) [inline wikiman.sh]: - This block shoud use `if-then-fi` syntax to follow the style.
- Print error to *STDERR*.
- Every fatal error has its own code.
- Single quotes for regular strings
```bash
if [ "$conf_fuzzy_finder" = 'fzf' ] || [ "$conf_fuzzy_finder" = 'sk' ]; then
	echo "error: $conf_fuzzy_finder is not compatible with the paramters used in this script" 1>&2
	exit 4
fi
```

File: wikiman.sh [hunk 2]
```diff
@@ -285,6 +296,23 @@ while getopts l:s:H:pqhRSk o; do
...
 
+# Dependency check
+
+echo 'Checking dependencies...'
+for dependency in 'man' 'fzf' 'rg' 'awk' 'w3m' "$conf_fuzzy_finder"; do
+	which "$dependency" >/dev/null || {
+		echo "Error: missing dependency: cannot find $dependency executable!"
+		exit 1
+	}
+done
```
Comment (filiparag) [inline wikiman.sh]: - Printing 'Checking dependencies...' every time is too verbose.
- Dependencies should be in an array defined beforehand.
- Style should be changed - see comment for lines 311-314.

**Review by filiparag — CHANGES_REQUESTED** [review]
Please make commented changes and add `-f` option to README.md.

File: wikiman.sh (hunk 2 above)
Comment (Z5483) [reply wikiman.sh:307]: Do you want the exit code in this block set to `127`?

Comment (Z5483) [reply wikiman.sh:307]: >      Printing 'Checking dependencies...' every time is too verbose.

This only get printed 1 time before the dependency check start. I don't understand exactly what I you want. Do you want it to be removed completely?

**Comment by khuei** [comment]
> Please make commented changes and add `-f` option to README.md.

I'm thinking about putting the dependency list to a separate section before the [INSTALLATION](https://github.com/filiparag/wikiman#installation). This would cause less confusion for the user to know which fuzzy finder is required.

File: wikiman.sh (hunk 1 above)
Comment (khuei) [reply wikiman.sh]: > This block shoud use `if-then-fi` syntax to follow the style.

The check would not work with the `if-then-fi` syntax, so I change it to `case-esac` syntax instead.

**Review by filiparag — APPROVED** [review]

**Comment by filiparag** [comment]
Your changes can now be accepted.
Thanks for contributing to Wikiman.

**Comment by khuei** [comment]
Thank you for the merge!

**Pull request merged**